packages/esl/src/esl-popup/core/esl-popup-position.ts

```diff
@@ -98,10 +103,20 @@ function calcPopupPositionByMajorAxis(cfg: PopupPositionConfig): number {
  * @param cfg - popup position config
  */
 function calcPopupPositionByMinorAxis(cfg: PopupPositionConfig): number {
-  const {position, inner, arrow, marginArrow, offsetArrowRatio} = cfg;
-  const centerPosition = inner[isOnHorizontalAxis(position) ? 'cy' : 'cx'];
-  const dimensionName = getDimensionName(position, true);
-  return centerPosition - arrow[dimensionName] / 2 - marginArrow - calcUsableSizeForArrow(cfg, dimensionName) * offsetArrowRatio;
+  const {placement, alignment, arrow, offsetTetherRatio, marginTether, trigger} = cfg;
+  let tetherPlacement;
+  switch (alignment) {
+    case 'start':
+      tetherPlacement = trigger[isOnHorizontalAxis(placement) ? 'y' : 'x'];
+      break;
+    case 'end':
+      tetherPlacement = trigger[isOnHorizontalAxis(placement) ? 'bottom' : 'right'];
+      break;
+    default:
+      tetherPlacement = trigger[isOnHorizontalAxis(placement) ? 'cy' : 'cx'];
+  }
```

Comment:
Cosmetic: To simplify that function and avoid writable variable declarations, we can introduce a separate method for tetherPlacement calculation.